src/stores/chat.ts

```diff
@@ -196,6 +196,41 @@ export const useChatStore = defineStore('chat', () => {
 		return Math.min(...filterNumericIds(contextBlock))
 	}
 
+	/**
+	 *
+	 * Check whether there are enough messages to render in the selected context, in particular
+	 * when there are other blocks, it means there is likely more history to load
+	 *
+	 * @param token The conversation token
+	 * @param data The data object containing messageId and threadId
+	 * @param data.messageId The message id
+	 * @param data.threadId The thread id
+	 */
+	function hasEnoughMessages(
+		token: string,
+		{ messageId = 0, threadId = 0 }: GetMessagesListOptions = { messageId: 0, threadId: 0 },
+	): boolean {
+		let contextBlock: Set<number>
+		let numBlocks = 0
```

Comment:
I would put lighter checks upfront. Unless there is a reason to not check it for a thread

```suggestion
		const numBlocks = (threadId ? threadBlocks[token][threadId]?.length : chatBlocks[token]?.length) ?? 0
		if (numBlocks <= 1) {
			// If only one block, we cannot assume there is more history to load
			return true
		}

		let contextBlock: Set<number>
```